lua/acf/hooks/hooks_sv.lua

```diff
@@ -0,0 +1,239 @@
+local Hooks = ACF.Utilities.Hooks
+
+
+Hooks.Add("ACF_Base_Server", function(Gamemode)
+	--- Called when the player has properly loaded onto the server.  
+	--- It's possible to use network messages in this hook, unlike PlayerInitialSpawn.
+	--- @param Player entity The player entity that just finished loading.
+	function Gamemode:ACF_OnLoadPlayer()
+	end
+
+	--- Called when the bullet will attempt to use the default flight behavior.  
+	--- All the values on the bullet are still updated towards the next position, even if this hook returns false.  
+	--- To prevent this, set Bullet.HandlesOwnIteration to true.
+	--- @param Bullet table The bullet object that will attempt to fly in the current tick.
+	--- @return boolean # False if the bullet shouldn't run the default flight behavior in the currect tick.
+	function Gamemode:ACF_PreBulletFlight()
+		return true
+	end
+
+	--- Called when attempting to determine if a bullet should ignore a particular entity.
+	--- @param Entity entity The entity hit by the bullet.
+	--- @param Bullet table The bullet being tested.
+	--- @return boolean # Return false if the bullet should filter out the entity.
+	function Gamemode:ACF_OnFilterBullet()
+		return true
+	end
+
+	--- Called after the default legality checks have failed to mark the entity as illegal.  
+	--- It is advised to not return true to prevent conflict between everything that uses this function.
+	--- @param Entity entity The entity that is currectly being checked for legality.
+	--- @return boolean # True if the entity is legal, false otherwise.
+	--- @return string # The reason why the entity is illegal. Not required if legal.
+	--- @return string # A short explanation on why the entity is illegal. Not required if legal.
+	--- @return number # Optionally, the amount of time in seconds the entity will remain illegal for. Not required if legal.
+	function Gamemode:ACF_OnCheckLegal()
+		return true
+	end
+
+	--- Called before an ACF entity is attempted to be spawned.  
+	--- This will only be called if the class has been registered and has a spawn function assigned to it.
+	--- @param Class string The entity class that is trying to be spawned.
+	--- @param Player entity The player attempting to spawn the entity.
+	--- @param Position vector The position where the entity is attempting to be spawned.
+	--- @param Angles angle The angles at which the entity is attempting to be spawned.
+	--- @param Data table A table with all the information required for the entity to set itself up.
+	--- @return boolean # True if the entity can be spawned, false otherwise.
+	--- @return string # A short explanation on why the entity can't be spawned. Not required if the entity is spawned.
+	function Gamemode:ACF_CanCreateEntity()
```

Comment:
Is `ACF_CanCreateEntity`/`ACF_CanUpdateEntity` having the word Can at the front but everything else having it in the middle of the phrase intentional or just an oversight?

Reply:
It's an oversight, probably from the time I was trying to copycat Gmod hook names.

Reply:
After a bit of further review, `ACF_CanUpdateEntity` may potentially have a naming conflict with `ACF_PreEntityUpdate`, as both would seemingly be named `ACF_PreUpdateEntity` under the new convention. These two hooks may need to be more meaningfully differentiated from each other somehow.